openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/EncounterResource.java

```diff
@@ -39,43 +40,34 @@
  * point in time (referred to as the "snapshot time").
  * @see AbstractReadOnlyResource
  */
-// TODO: Merge with PatientResource and let clients use a query parameter to
-// indicate whether encounter data should be included with each returned patient.
 @Resource(name = RestController.REST_VERSION_1_AND_NAMESPACE + "/encounters",
     supportedClass = Patient.class, supportedOpenmrsVersions = "1.10.*,1.11.*")
 public class EncounterResource
-    extends AbstractReadOnlyResource<Patient> implements Creatable {
+    extends AbstractReadOnlyResource<Encounter> implements Creatable {
     private final PatientService patientService;
-    private final EncounterService encounterService;
+    private final ProjectBuendiaService buendiaService;
 
     public EncounterResource() {
-        super("patient", Representation.DEFAULT);
+        super("encounter", Representation.DEFAULT);
         patientService = Context.getPatientService();
-        encounterService = Context.getEncounterService();
+        buendiaService = Context.getService(ProjectBuendiaService.class);
     }
 
     /**
-     * Returns all patients.  The retrieved records will be filled in with
-     * each patient's encounter and observation data by
-     * {@link #populateJsonProperties(Patient, RequestContext, SimpleObject, long)}
-     * on its way to becoming JSON that is sent to the client.
-     * @param context      unused here; see populateJsonProperties() for details
+     * Returns all encounters observed after the "since" parameter, or since the beginning of time
+     * if the "since" parameter is not set.
+     * @param context      used to obtain the "since" parameter for incremental sync.
      * @param snapshotTime unused here; see populateJsonProperties() for details
      * @see AbstractReadOnlyResource#search(RequestContext)
      */
-    @Override public List<Patient> searchImpl(RequestContext context, long snapshotTime) {
-        String patientUuid = context.getParameter("patient_uuid");
-        if (patientUuid != null) {
-            Patient patient = patientService.getPatientByUuid(patientUuid);
-            if (patient == null) {
-                throw new ObjectNotFoundException();
-            }
-            List<Patient> patients = new ArrayList<>();
-            patients.add(patient);
-            return patients;
-        } else {
-            return patientService.getAllPatients();
+    @Override public List<Encounter> searchImpl(RequestContext context, long snapshotTime) {
+        Date syncFrom;
+        try {
+            syncFrom = RequestUtil.getSyncFromDate(context);
```

Comment:
The Calendar API was the one I had in mind (search for "nextSyncToken").

Sorry, I thought I had seen this as a common technique in a few APIs, but
now I can only find it in the Calendar API!
On Nov 2, 2015 8:32 PM, "Fabian Tamp" <notifications@github.com> wrote:

> In
> openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/EncounterResource.java
> <https://github.com/projectbuendia/buendia/pull/20#discussion_r43715476>:
>
> > -    @Override public List<Patient> searchImpl(RequestContext context, long snapshotTime) {
> > -        String patientUuid = context.getParameter("patient_uuid");
> > -        if (patientUuid != null) {
> > -            Patient patient = patientService.getPatientByUuid(patientUuid);
> > -            if (patient == null) {
> > -                throw new ObjectNotFoundException();
> > -            }
> > -            List<Patient> patients = new ArrayList<>();
> > -            patients.add(patient);
> > -            return patients;
> > -        } else {
> > -            return patientService.getAllPatients();
> > +    @Override public List<Encounter> searchImpl(RequestContext context, long snapshotTime) {
> > +        Date syncFrom;
> > +        try {
> > +            syncFrom = RequestUtil.getSyncFromDate(context);
>
> Which Google APIs are you talking about?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/projectbuendia/buendia/pull/20/files#r43715476>.
>